internal/charmstore/migrations.go

```diff
@@ -222,3 +227,73 @@ func populateWriteACL(db StoreDatabase) error {
 	logger.Infof("%d base entities updated", counter)
 	return nil
 }
+
+func populatePromulgatedEntities(db StoreDatabase) error {
+	entities := db.Entities()
+	iter := entities.Find(bson.D{{"user", ""}}).Select(bson.D{
+		{"_id", 1},
+		{"baseurl", 1},
+		{"blobhash", 1},
+		{"extrainfo.bzr-owner", 1},
+	}).Sort("revision").Iter()
+	var entity mongodoc.Entity
+	for iter.Next(&entity) {
+		logger.Debugf("URL: %s", entity.URL)
+		var user string
+		if err := json.Unmarshal([]byte(entity.ExtraInfo["bzr-owner"]), &user); err != nil {
+			return errgo.Notef(err, "cannot unmarshal user from extra-info")
```

Comment:
I wonder if we should really abort the migration if we can't find the bzr owner. Perhaps print a log message and use "unknown" (or something else suitable) as the user in this case, perhaps?